source/common/router/config_impl.cc

```diff
@@ -640,14 +685,53 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
   }
 
   if (route.route().has_regex_rewrite()) {
-    if (!prefix_rewrite_.empty()) {
-      throw EnvoyException("Cannot specify both prefix_rewrite and regex_rewrite");
+    if (!prefix_rewrite_.empty() || path_match_policy_.enabled()) {
+      throw EnvoyException("Specify only one of prefix_rewrite, regex_rewrite or path_rewrite_policy");
     }
     auto rewrite_spec = route.route().regex_rewrite();
     regex_rewrite_ = Regex::Utility::parseRegex(rewrite_spec.pattern());
     regex_rewrite_substitution_ = rewrite_spec.substitution();
   }
 
+  if (path_match_policy_.enabled()) {
+    if (!prefix_rewrite_.empty() || route.route().has_regex_rewrite()) {
+      throw EnvoyException("Specify only one of prefix_rewrite, regex_rewrite or path_rewrite_policy");
+    }
+  }
+
+  if (path_match_policy_.enabled()) {
+    // TODO: validate path_match_policy is of kind the pattern template
+    if (!Extensions::PatternTemplate::isValidMatchPattern(path_match_policy_.predicate()->pattern())
+             .ok()) {
+      throw EnvoyException(fmt::format("path_match_policy {} is invalid",
+                                       path_match_policy_.predicate()->pattern()));
+    }
+  }
+
+  if (path_rewrite_policy_.enabled()) {
+    // TODO: validate path_rewrite_policy is of kind the pattern template
+    if (!Extensions::PatternTemplate::isValidPathTemplateRewritePattern(
+             path_rewrite_policy_.predicate()->pattern())
+             .ok()) {
+      throw EnvoyException(fmt::format("path_rewrite_policy {} is invalid",
+                                       path_match_policy_.predicate()->pattern()));
+    }
+
+    if (!Extensions::PatternTemplate::isValidSharedVariableSet(
+             path_rewrite_policy_.predicate()->pattern(), path_match_policy_.predicate()->pattern())
+             .ok()) {
+      throw EnvoyException(fmt::format(
+          "mismatch between path_match_policy {} and path_rewrite_policy {}",
+          path_match_policy_.predicate()->pattern(), path_rewrite_policy_.predicate()->pattern()));
+    }
+  }
+
+  if (!prefix_rewrite_.empty()) {
+    if (route.route().has_regex_rewrite() || path_match_policy_.enabled()) {
```

Comment:
path_match vs path_rewrite here too.

Reply:
I wonder if we shouldn't gather the 3 places where the mutual exclusion checks occur into one area at the top? I wish there was a `std::one_of` to make it a simple if.

Reply:
Moved into single location at top